lib/sycamore/sycamore/docset.py

```diff
@@ -1523,6 +1529,31 @@ def execute(self, **kwargs) -> None:
         """
 
         from sycamore.executor import Execution
+        from sycamore.materialize import Materialize
 
-        for doc in Execution(self.context).execute_iter(self.plan, **kwargs):
-            pass
+        if isinstance(self.plan, Materialize) and self.plan._reliability is not None:
+            mrr = self.plan._reliability
+
+            assert self.plan.child().last_node_materialize(), "first step must be materialize step"
+
+            while True:
```

Comment:
Minor, but need a hook here to call before each iteration, ideally with an iteration count and the previous round error so that we can clear the jupyter output between rounds, and print out status information (count & prev error).

Without that, too much output makes jupyter wedge up.